lib/manageiq/api/common/user.rb

```diff
@@ -0,0 +1,69 @@
+module ManageIQ
+  module API
+    module Common
+      class User
+        IDENTITY_KEY = 'x-rh-identity'.freeze
+
+        def self.current
+          ManageIQ::API::Common::Request.current
+        end
+
+        def username
+          validate_and_return(__method__)
+        end
+
+        def email
+          validate_and_return(__method__)
+        end
+
+        def first_name
+          validate_and_return(__method__)
+        end
+
+        def last_name
+          validate_and_return(__method__)
+        end
+
+        def is_active?
+          validate_and_return(:is_active)
+        end
+
+        def is_org_admin?
+          validate_and_return(:is_org_admin)
+        end
+
+        def is_internal?
+          validate_and_return(:is_internal)
+        end
+
+        def locale
+          validate_and_return(__method__)
+        end
+
+        def tenant
+          result = decode.dig('identity', 'account_number')
+          raise StandardError, "Tenant key doesn't exist" if result.nil?
+          result
+        end
+
+        private
+
+        def decode
+          hashed = ManageIQ::API::Common::Request.current.to_h
+          user_hash = hashed[:headers][IDENTITY_KEY]
+          JSON.parse(Base64.decode64(user_hash))
```

Comment:
Do we have to actual decode the user_hash every time since decode is so often called in this class? We may cache the decoded result.
There is a possibility that current user and current request are out of sync by caching the result, but the real use case for current user is almost within the context of current request.

Reply:
+1.  This should memoize the decoding because it's using the "current" header (and for performance).

Reply:
Memoization will be added in a follow up PR as per this issue and conversations with @Fryguy: https://github.com/ManageIQ/manageiq-api-common/issues/36